crates/gradbench/src/main.rs

```diff
@@ -1054,6 +1064,16 @@ fn log_command(command: LogCommands) -> anyhow::Result<()> {
                 Ok(())
             }
         },
+        LogCommands::Summary { input } => match input {
+            Some(input_path) => {
+                log::summary(&mut io::BufReader::new(fs::File::open(input_path)?))?;
+                Ok(())
+            }
+            None => {
+                log::summary(&mut io::BufReader::new(io::stdin()))?;
+                Ok(())
+            }
+        },
```

Comment:
I'm guessing in the future we could make a utility function that does this one-time dispatching automatically. Or, alternatively, possibly a better design could be to have the `summary` function just take an `Iterator` of log lines. Again, no need to change it in this PR, I'm just thinking about stuff.